met-web/src/components/engagement/view/EngagementBanner/StandAloneBanner.tsx

```diff
@@ -6,16 +6,17 @@ import { getEngagement } from 'services/engagementService';
 
 interface EngagementBannerProps {
     startSurvey: () => void;
-    engagementSlug: string;
+    engagementUrl: string;
 }
-export const EngagementBanner = ({ startSurvey, engagementSlug }: EngagementBannerProps) => {
+export const EngagementBanner = ({ startSurvey, engagementUrl }: EngagementBannerProps) => {
     const [isEngagementLoading, setIsEngagementLoading] = useState(true);
     const [savedEngagement, setSavedEngagement] = useState<Engagement | null>(null);
     const [engagementId, setEngagementId] = useState<number | null>(null);
     const [isError, setIsError] = useState(false);
 
     const loadEngagementSlug = async () => {
         try {
+            const engagementSlug = engagementUrl.substring(engagementUrl.lastIndexOf('/') + 1, engagementUrl.length);
```

Comment:
any chance we can send it the engagementSlug from outside rather than calculating it here. Just for the sake of not coupling it with the url. With the engagementSlug sent to it there's less coupling I think